OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m

```diff
@@ -163,8 +164,9 @@ - (NSString *)getBucketingId:(NSString *)userId
     NSString *bucketingId = userId;
     // If the bucketing ID key is defined in attributes, then use that
     // in place of the userID for the murmur hash key
-    if (![OPTLYDecisionService isEmptyString:attributes[OptimizelyBucketId]]) {
-        bucketingId = attributes[OptimizelyBucketId];
+    
```

Comment:
Don't we need nil checking for "attributes" since it's changed nullable?
Why using for loop instead of a simple nil checking if statement?